sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1159,6 +1161,25 @@ class Analyzer(
       }
     }
   }
+
+  /**
+   * Removes natural joins.
```

Comment:
I think we need more comments here, how we resolve a natural join to normal join?